docker/sparkswapd/Dockerfile

```diff
@@ -1,79 +1,47 @@
-#
-#############################################
+RUN apk update
+RUN apk --no-cache add bash ca-certificates g++ gcc git make openssl python wget && \
+    wget -q -O /etc/apk/keys/sgerrand.rsa.pub https://alpine-pkgs.sgerrand.com/sgerrand.rsa.pub && \
+    wget https://github.com/sgerrand/alpine-pkg-glibc/releases/download/2.28-r0/glibc-2.28-r0.apk && \
+    apk add glibc-2.28-r0.apk
 
-ARG KEY_PATH
-RUN : "${KEY_PATH:?KEY_PATH Build argument needs to be set.}"
-ARG CERT_PATH
-RUN : "${CERT_PATH:?CERT_PATH Build argument needs to be set.}"
-
-ARG TLS_KEY="/secure/broker-rpc-tls.key"
-ARG TLS_CERT="/secure/broker-rpc-tls.cert"
-
-COPY ${KEY_PATH} ${TLS_KEY}
-COPY ${CERT_PATH} ${TLS_CERT}
-
-#############################################
-# Keypair Generation for Relayer
-#
-# This step creates certs to allow the broker to authenticate/auth for orders
-# on the relayer
-#
-# We use a "Secure key exchange algorithm" here because these keys are exchanged
-# via a non secure channel. (ECDH)
-#
-#############################################
-
-ARG ID_PRIV_KEY='/secure/broker-identity.private.pem'
-ARG ID_PUB_KEY='/secure/broker-identity.public.pem'
-
-RUN openssl ecparam -name prime256v1 -genkey -noout -out ${ID_PRIV_KEY}
-RUN openssl ec -in ${ID_PRIV_KEY} -pubout -out ${ID_PUB_KEY}
-
-#############################################
-# Broker installation steps
-#############################################
-
-# Copy local broker files to container
-COPY . /home/app
+WORKDIR /home/app
 
-# Download relayer proto and add it to container
-ARG RELAYER_PROTO_BRANCH
-ARG RELAYER_PROTO_COMMIT_SHA
+RUN git clone -b ${BROKER_VERSION} https://github.com/sparkswap/broker.git .
 
-RUN mkdir -p /home/app/proto
-RUN git clone --branch ${RELAYER_PROTO_BRANCH} https://github.com/sparkswap/relayer-proto.git /home/app/proto \
-&&  (cd /home/app/proto && git checkout ${RELAYER_PROTO_COMMIT_SHA})
+RUN npm run build -- --no-docker --production
 
+FROM node:8.11-alpine as final
 
-# Start a new, final image to reduce size.
-FROM node:8.11 as final
+RUN apk update
+RUN apk add bash curl
 
 RUN mkdir "/secure"
 
-# Copy certs
-COPY --from=builder /secure /secure
-
 # Copy the application
-COPY --from=builder /home/app /home/app
+COPY --from=builder /home/app/broker-daemon /home/app/broker-daemon
+COPY --from=builder /home/app/broker-cli /home/app/broker-cli
+COPY --from=builder /home/app/scripts/start-sparkswapd.sh /home/app/scripts/start-sparkswapd.sh
+COPY --from=builder /home/app/package.json /home/app/package.json
+COPY --from=builder /home/app/pm2.json /home/app/pm2.json
+
+# Copy certs
+COPY --from=builder /home/app/certs /secure
 
 # Copy relayer-proto files
 COPY --from=builder /home/app/proto /home/app/proto
 
+# Copy node_modules and rebuild
+COPY --from=builder /home/app/node_modules ./node_modules
+
 WORKDIR /home/app
 
-RUN npm install --quiet
-RUN npm run broker-proto
+RUN npm rebuild -q
+RUN npm install pm2 -g
```

Comment:
@treygriffith this is a good point. we should move PM2 to dependencies in `package.json` instead of dev dependencies (we use --production for npm above)

Reply:
@treygriffith this was actually a huge bug. I wasn't copying `node_modules` to the correct folder in the docker image which is why `pm2` was not present (and required me to globally install pm2). Thanks for the catch